tests/Loader/CucumberNDJsonAstLoaderTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\Behat\Gherkin\Loader;
+
+use Behat\Gherkin\Exception\NodeException;
+use Behat\Gherkin\Loader\CucumberNDJsonAstLoader;
+use Behat\Gherkin\Node\BackgroundNode;
+use Behat\Gherkin\Node\ExampleTableNode;
+use Behat\Gherkin\Node\FeatureNode;
+use Behat\Gherkin\Node\OutlineNode;
+use Behat\Gherkin\Node\ScenarioNode;
+use PHPUnit\Framework\TestCase;
+
+final class CucumberNDJsonAstLoaderTest extends TestCase
+{
+    private CucumberNDJsonAstLoader $loader;
+
+    protected function setUp(): void
+    {
+        $this->loader = new CucumberNDJsonAstLoader();
+    }
+
+    public function testStringResourcesAreSupported(): void
+    {
+        $this->assertTrue($this->loader->supports('a string'));
+    }
+
+    public function testValidLoading(): void
```

Comment:
I think most of the points here are already covered by the external cucumber parity tests, but I don't think it's a bad idea to have this here. It could serve as a blueprint/inspiration for testing other situations.